src/decorator/options/ColumnOptions.ts

```diff
@@ -117,6 +118,11 @@ export interface ColumnOptions extends ColumnCommonOptions {
      */
     asExpression?: string;
 
+    /**
+     * Like asExpression, but support all database in typeorm.
```

Comment:
The type parameter can avoid to use global variables to determine database type and help to encapsulate code. If you don't consider the compatibility when defining the model, just ignore this parameter @Kononnable

Reply:
I don't really see use of it in real application. It looks to me like it would be used only to show that tests are working in multiple database types(but test column could also do something simple like `1+1` and it would be easier to read).

Reply:
It can let you encapsulate a group of sql generating function and don't consider what the database type when using they, such as  the `concat` and `summary` function in my test cases. if haven't the parameter, you need write function like `mysql_concat`, `pg_concat`, `mssql_concat`, etc, that is very inconvenience and error-prone, and the `concat` is common scenarios.